code/modules/mob/living/carbon/life.dm

```diff
@@ -201,3 +207,61 @@
 /mob/living/carbon/proc/add_slowdown(amount)
 	slowdown = adjust_slowdown(amount*STANDARD_SLOWDOWN_REGEN)
 	return slowdown
+
+/mob/living/carbon/proc/breathe()
+	if(!need_breathe())
+		return
+
+	if(health < get_crit_threshold() && !reagents.has_reagent("inaprovaline"))
+		Losebreath(1, TRUE)
+	else
+		adjust_Losebreath(-1, TRUE)
+
+	if(!losebreath)
+		. = get_breath_from_internal()
+		if(!.)
+			. = get_breath_from_environment()
+
+	handle_breath(.)
+
+/mob/living/carbon/proc/get_breath_from_internal()
+	if(internal)
+		if(istype(buckled,/obj/machinery/optable))
+			var/obj/machinery/optable/O = buckled
+			if(O.anes_tank)
+				return O.anes_tank.return_air()
+		if(!contents.Find(internal))
+			internal = null
+		if(!wear_mask || !(wear_mask.flags_inventory & ALLOWINTERNALS))
+			internal = null
+		if(internal)
+			hud_used.internals.icon_state = "internal1"
+			return internal.return_air()
+		else if(hud_used?.internals)
+			hud_used.internals.icon_state = "internal0"
+
+/mob/living/carbon/proc/get_breath_from_environment()
+	if(istype(loc, /atom/movable))
+		var/atom/movable/container = loc
+		. = container.handle_internal_lifeform(src)
+
+	else if(isturf(loc))
```

Comment:
Make this check come first, as it will account for the vast majority of the cases.